lua/entities/gmod_wire_expression2/core/custom/cfc_entity.lua

```diff
@@ -52,4 +52,21 @@ e2function void entity:ejectPodTo(vector pos)
 
     driver:ExitVehicle()
     driver:SetPos(clampedPos)
-end
\ No newline at end of file
+end
+
+
+-- Easy way to have some of the setPos benefits without mostly being able to be abused.
+e2function number entity:setFairPos(vector pos)
+    if this:IsPlayer() then return 3 end -- Debug output 3 if the target is a player, because we dont want someone tping themselves around the e2 or any other player.
+    local posToLoc = pos - self:GetPos() -- distance to the wanted position
+    local lengthPos = posToLoc[1]^2 + posToLoc[2]^2 + posToLoc[3]^2 -- we dont need to do the square root on this, itll just waste time and resources
+
+    local entB4PosLoc = this:GetPos() - self:GetPos() -- get distance to target entity
+    local entLength = entB4PosLoc[1]^2 + entB4PosLoc[2]^2 + entB4PosLoc[3]^2 -- we dont need to do the square root on this, itll just waste time and resources
+
+    if lengthPos >= CFCMaxFairSetDist^2 then return 1 end -- Debug output 1 if the wanted pos is out of bounds and end
+    if entLength >= CFCMaxFairSetDist^2 then return 2 end -- Debug output 2 if the target entity pos is out of bounds and end
```

Comment:
Same comment as before. `CFCMaxFairSetDist^2` should be its own var so the comparison is self-explanatory.

`nil` or `false` is a sufficient return here.